examples/bicycle/whipple.py

```diff
@@ -0,0 +1,508 @@
+Id = mec.inertia(C, id11, id22, id11, 0.0, 0.0, 0.0)
+Ie = mec.inertia(E, ie11, ie22, ie33, 0.0, 0.0, ie31)
+If = mec.inertia(E, if11, if22, if11, 0.0, 0.0, 0.0)
+
+##############
+# Rigid Bodies
+##############
+
+print('Defining the rigid bodies.')
+
+rear_frame = mec.RigidBody('Rear Frame', co, C, mc, (Ic, co))
+rear_wheel = mec.RigidBody('Rear Wheel', do, D, md, (Id, do))
+front_frame = mec.RigidBody('Front Frame', eo, E, me, (Ie, eo))
+front_wheel = mec.RigidBody('Front Wheel', fo, F, mf, (If, fo))
+
+bodies = [rear_frame, rear_wheel, front_frame, front_wheel]
+
+###########################
+# Generalized Active Forces
+###########################
+
+print('Defining the forces and torques.')
+
+# gravity
+Fco = (co, mc * g * A['3'])
+Fdo = (do, md * g * A['3'])
+Feo = (eo, me * g * A['3'])
+Ffo = (fo, mf * g * A['3'])
+
+# input torques
+Tc = (C, T4 * A['1'] - T6 * B['2'] - T7 * C['3'])
+Td = (D, T6 * C['2'])
+Te = (E, T7 * C['3'])
+
+forces = [Fco, Fdo, Feo, Ffo, Tc, Td, Te]
+
+###############
+# Kane's Method
+###############
+
+print("Generating Kane's equations.")
+
+# var   : J , G
+# yaw   : q3, q1
+# roll  : q4, q2
+# pitch : q5, q4
+# steer : q7, q5
+# rwh   : u6, u3
+# fwh   : u8, u6
+
+kane = mec.KanesMethod(N,
+                       [q3, q4, q7],  # yaw, roll, steer
+                       [u4, u6, u7],  # roll rate, rear wheel rate, steer rate
+                       kd_eqs=kinematical,
+                       q_dependent=[q5],  # pitch angle
+                       configuration_constraints=[holonomic],
+                       u_dependent=[u3, u5, u8],  # yaw rate, pitch rate, front wheel rate
+                       velocity_constraints=nonholonomic)
+
+fr, frstar = kane.kanes_equations(forces, bodies)
+
+mass_matrix = kane.mass_matrix
+print('The mass matrix is a function of these dynamic variables:')
+print(list(sm.ordered(mec.find_dynamicsymbols(mass_matrix))))
+
+# sub in the kin diffs to eliminate some extraneous derivatives
+forcing_vector = mec.msubs(kane.forcing, kane.kindiffdict())
+print('The forcing vector is a function of these dynamic variables:')
+print(list(sm.ordered(mec.find_dynamicsymbols(forcing_vector))))
+
+# NOTE : doing this substitution must make the expressions super long
+# because it causes all kinds of slowdowns, for example cse is slow in the
+# code gen step.
+# u3, u5, u8 are all in forcing_vector and need to be replaced by functions
+# of u4, u6, u7
+# These are the equations for u3, u5, u8 in terms of the independent speeds:
+#u_dep = kane._Ars * kane.u[:3, :]
+#print('The dependent speeds are a function of these dynamic variables:')
+#print(list(sm.ordered(mec.find_dynamicsymbols(u_dep))))
+#
+#forcing_vector = mec.msubs(forcing_vector, dict(zip([u3, u5, u8], u_dep)))
+# This takes forever.
+#print(list(sm.ordered(mec.find_dynamicsymbols(forcing_vector))))
+
+####################################
+# Validation of non-linear equations
+####################################
+
```

Comment:
One thing to check: how am I handling the holonomic constraint in the numerical evaluation? It isn't clear if I am.